src/piratechain/piratechainTypes.ts

```diff
@@ -25,83 +36,18 @@ export interface PiratechainNetworkInfo {
   transactionQueryLimit: number
 }
 
-export interface PiratechainSpendInfo {
-  zatoshi: string
-  toAddress: string
-  memo: string
-  fromAccountIndex: number
-  mnemonicSeed: string
-}
-
-export interface PiratechainTransaction {
-  rawTransactionId: string
-  blockTimeInSeconds: number
-  minedHeight: number
-  value: string
-  toAddress?: string
-  memos: string[]
-}
-
-export interface PiratechainPendingTransaction {
-  txId: string
-  raw: string
-}
-
-export interface PiratechainWalletBalance {
-  availableZatoshi: string
-  totalZatoshi: string
-}
-
-export interface PiratechainInitializerConfig {
-  networkName: PiratechainNetworkName
-  defaultHost: string
-  defaultPort: number
-  mnemonicSeed: string
-  alias: string
-  birthdayHeight: number
-}
-
-export interface PiratechainAddresses {
-  // unifiedAddress: string
-  saplingAddress: string
-  // transparentAddress: string
-}
-
-export type PiratechainSynchronizerStatus =
-  | 'STOPPED'
-  | 'DISCONNECTED'
-  | 'SYNCING'
-  | 'SYNCED'
-
-export interface PiratechainStatusEvent {
-  alias: string
-  name: PiratechainSynchronizerStatus
-}
-
-export interface PiratechainUpdateEvent {
-  alias: string
-  isDownloading: boolean
-  isScanning: boolean
-  lastDownloadedHeight: number
-  lastScannedHeight: number
-  scanProgress: number // 0 - 100
-  networkBlockHeight: number
-}
-
-// Block range is inclusive
-export const asPiratechainBlockRange = asObject({
-  first: asNumber,
-  last: asNumber
-})
```

Comment:
You can keep this and pass the generic type argument to `asObject<BlockRange>` to ensure the cleaner matches the type.